distributed/client.py

```diff
@@ -1308,34 +1317,55 @@ async def _update_scheduler_info(self):
         except OSError:
             logger.debug("Not able to query scheduler for identity")
 
-    async def _wait_for_workers(self, n_workers=0, timeout=None):
+    async def _wait_for_workers(
+        self, n_workers=0, timeout=None, mode=WorkerWaitMode.min
+    ):
+        if isinstance(mode, str):
+            mode = WorkerWaitMode(mode)
         info = await self.scheduler.identity()
         self._scheduler_identity = SchedulerInfo(info)
         if timeout:
             deadline = time() + parse_timedelta(timeout)
         else:
             deadline = None
 
-        def running_workers(info):
+        def running_workers(info, status_list=[Status.running]):
             return len(
                 [
                     ws
                     for ws in info["workers"].values()
-                    if ws["status"] == Status.running.name
+                    if ws["status"] in [s.name for s in status_list]
                 ]
             )
 
-        while n_workers and running_workers(info) < n_workers:
+        need_exact = lambda: (
+            running_workers(info, status_list=[Status.running, Status.paused])
+            != n_workers
+            and mode == WorkerWaitMode.exactly
+        )
+        need_min = lambda: (
+            n_workers
+            and running_workers(info) < n_workers
+            and mode == WorkerWaitMode.min
+        )
+        need_max = lambda: (
+            n_workers
+            and running_workers(info, status_list=[Status.running, Status.paused])
+            > n_workers
+            and mode == WorkerWaitMode.max
+        )
+
+        while need_exact() or need_min() or need_max():
```

Comment:
Personally, I would find this to be clearer if we first decided on a stop conditon based on the mode and then just used that in the loop. Similar to this:
```suggestion
        if mode is WorkerWaitMode.min:
            stop_condition = lambda: running_workers(info) >= n_workers
        elif mode is WorkerWaitMode.exactly:
            stop_condition = lambda: running_workers(info, status_list=[Status.running, Status.paused]) == n_workers
        elif mode is WorkerWaitMode.max:
            stop_condition = lambda: running_workers(info, status_list=[Status.running, Status.paused]) <= n_workers
        else:
            raise NotImplementedError(f"{mode} is not handled.")

        while not stop_condition():
```

YMMV though, I'll leave the decision up to you.